lib/connectors/memory.js

```diff
@@ -54,10 +55,38 @@ Memory.prototype.getTypes = function() {
 };
 
 Memory.prototype.connect = function(callback) {
+  var self = this;
+  if (this.connected) {
+    // It's already connected or being connected
+    // Do NOT run loadFromFile again as it will override the cache
+    if (callback) process.nextTick(callback);
+    return;
+  }
+  if (callback) {
+    // Queue the callback
+    this.pendingConnectCallbacks = this.pendingConnectCallbacks || [];
+    this.pendingConnectCallbacks.push(callback);
+  }
+  // The connect is already in progress
+  if (this.connecting) return;
+  // Set connecting flag to be true
+  this.connecting = true;
   if (this.isTransaction) {
     this.onTransactionExec = callback;
   } else {
-    this.loadFromFile(callback);
+    this.loadFromFile(function(err) {
+      if (!err) self.connected = true;
+      self.connecting = false;
+      // Invoke all pending callbacks
+      async.each(self.pendingConnectCallbacks, function(cb, done) {
+        process.nextTick(function() {
+          cb(err);
+          done();
+        });
+      }, function(err) {
+        self.pendingConnectCallbacks = [];
```

Comment:
What if somebody pushed into pendingConnectCallbacks while we were processing the queue? I think the following approach should be safer:

```js
self.connecting = false;
const pendingCallbacks = self.pendingConnectCallbacks;
self.pendingConnectCallbacks = [];

async.each(pendingCallbacks, ...);
```